diff -ur dahdi-linux-2.3.0.orig//drivers/dahdi/Kbuild dahdi-linux-2.3.0/drivers/dahdi/Kbuild
--- dahdi-linux-2.3.0.orig//drivers/dahdi/Kbuild	2010-02-25 20:10:02.000000000 +0100
+++ dahdi-linux-2.3.0/drivers/dahdi/Kbuild	2010-04-16 14:08:59.205604749 +0200
@@ -1,5 +1,5 @@
 obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI)			+= dahdi.o
-#obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_DUMMY)		+= dahdi_dummy.o
+obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_DUMMY)		+= dahdi_dummy.o
 obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_DYNAMIC)		+= dahdi_dynamic.o
 obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_DYNAMIC_LOC)	+= dahdi_dynamic_loc.o
 obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_DYNAMIC_ETH)	+= dahdi_dynamic_eth.o
