--- dahdi-linux-3.1.0/include/dahdi/kernel.h~	2021-02-06 20:19:28.000000000 +0100
+++ dahdi-linux-3.1.0/include/dahdi/kernel.h	2021-02-06 20:27:57.430773118 +0100
@@ -1427,7 +1427,7 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)
 #define refcount_read atomic_read
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 10, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)
 #define dahdi_ktime_equal ktime_equal
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 0, 0)
