--- dahdi-linux-3.1.0/include/dahdi/kernel.h~	2020-10-21 07:44:12.000000000 +0200
+++ dahdi-linux-3.1.0/include/dahdi/kernel.h	2020-10-21 08:05:07.153002470 +0200
@@ -58,6 +58,13 @@
 
 #include <linux/poll.h>
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
+#define HAVE_UNLOCKED_IOCTL 1
+#ifdef CONFIG_COMPAT
+#define HAVE_COMPAT_IOCTL 1
+#endif
+#endif
+
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0)
 #ifdef CONFIG_PCI
 #include <linux/pci-aspm.h>
