--- dahdi-linux-3.0.0/include/dahdi/kernel.h~	2018-11-15 15:03:40.000000000 +0100
+++ dahdi-linux-3.0.0/include/dahdi/kernel.h	2019-03-10 20:14:29.249473418 +0100
@@ -1406,9 +1406,13 @@
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 15, 0)
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 14, 104)
 #ifndef TIMER_DATA_TYPE
 #define TIMER_DATA_TYPE unsigned long
 #endif
+#else
+#define TIMER_DATA_TYPE struct timer_list *
+#endif
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 13, 0)
 
