--- dahdi-linux-3.1.0/include/dahdi/kernel.h~	2022-04-02 22:49:13.000000000 +0200
+++ dahdi-linux-3.1.0/include/dahdi/kernel.h	2022-04-02 23:08:10.095243044 +0200
@@ -1504,6 +1504,10 @@
 
 #endif /* 4.15.0 */
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 17, 0)
+#define PDE_DATA pde_data
+#endif
+
 #ifndef dahdi_ktime_equal
 static inline int dahdi_ktime_equal(const ktime_t cmp1, const ktime_t cmp2)
 {
